commands/core/list.go

```diff
@@ -34,17 +35,20 @@ func GetPlatforms(instanceID int32, updatableOnly bool) ([]*cores.PlatformReleas
 		return nil, errors.New("invalid instance")
 	}
 
-	res := []*cores.PlatformRelease{}
+	res := []*rpc.Platform{}
 	for _, targetPackage := range packageManager.Packages {
 		for _, platform := range targetPackage.Platforms {
-			if platformRelease := packageManager.GetInstalledPlatformRelease(platform); platformRelease != nil {
-				if updatableOnly {
+			platformRelease := packageManager.GetInstalledPlatformRelease(platform)
+			if platformRelease != nil {
+				if req.UpdatableOnly {
 					if latest := platform.GetLatestRelease(); latest == nil || latest == platformRelease {
 						continue
 					}
 				}
 
-				res = append(res, platformRelease)
+				rpcPlatform := commands.PlatformReleaseToRPC(platformRelease)
+				rpcPlatform.Installed = platformRelease.Version.String()
```

Comment:
Shouldn't:

```go
rpcPlatform.Installed = platformRelease.Version.String()
```

be performed inside `commands.PlatformReleaseToRPC(...)` ?

Reply:
```
// PlatformReleaseToRPC converts our internal structure to the RPC structure.
// Note: this function does not touch the "Installed" field of rpc.Platform as it's not always clear that the
// platformRelease we're currently converting is actually installed.
func PlatformReleaseToRPC(platformRelease *cores.PlatformRelease) *rpc.Platform {
```
From: https://github.com/arduino/arduino-cli/blob/a9b0c9df7f8dca8cac87b5337c583b4510c5fc56/commands/core.go#L23-L26

I prefered leaving as it is instead of trying to fix and breaking some things.

Reply:
Ok, we probably need to refactor this function and change it to `PlatformToRPC(platform *cores.Platform) *rpc.Platform`.